Make achievement enhancements work on comparison pages

ge-ku opened the pull request:
Hello, this enables SteamDB achievement enhancements on id|profile/\*/stats/\*/compare pages such as https://steamcommunity.com/id/ChetFaliszek/stats/383870/compare

It looks like this
![image](https://github.com/user-attachments/assets/23aa0650-b4a7-49cd-afb4-5f1bbfe868b9)


I tested it a little bit and it seems to work fine unless I'm missing some edge cases.

It works with progress bar achievements, assuming that original text for the second player is always something like "(14,035,100 / 50,000,000)". Unfortunately there is no progress info for the second player if the first one got the achievement already.

![image](https://github.com/user-attachments/assets/e9f8a992-6458-480e-9d1e-7480385de841)

xPaw commented:
Seems mostly ok, just this second progress bar text is a bit misaligned and the color is really muted:

![image](https://github.com/user-attachments/assets/fafa341d-5362-406f-9298-277e749eb9fe)

compared to colors here:

![image](https://github.com/user-attachments/assets/7bf26772-033b-47fb-9db6-272aef2d4ab0)

xPaw commented on scripts/community/achievements.js:
```diff
@@ -927,6 +1080,10 @@ function InitAchievements( items, isPersonal )
 			document.querySelector( '#topSummaryAchievements .achieveBar' )?.classList.add( 'steamdb_achievement_progressbar' );
+			if( isCompareView )
+			{
+				document.querySelector( '#topSummaryAchievements :nth-child(2 of .achieveBar)' )?.classList.add( 'steamdb_achievement_progressbar' );
```
Just querySelectorAll( '#topSummaryAchievements .achieveBar' ) and add a class to them all?

xPaw commented on scripts/community/achievements.js:
```diff
@@ -120,21 +120,34 @@ function InitAchievements( items, isPersonal )
+		if( window.innerWidth < 600 )
+		{
+			extraTabs.classList.add( 'steamdb_stats_extra_tabs_mobile' );
+			document
```
If this is to be run on every resize, it should at least check that the element isn't already where it should be, recreating it is not good.

xPaw commented on styles/achievements.css:
```diff
@@ -301,6 +321,14 @@ html {
 	width: 0%;
 }
 
+:nth-child(2 of .steamdb_achievement_progressbar) .achieveBarProgress {
```
Use `steamdb_achievement_progress_compare` class

xPaw commented:
Seems like there's a bug with 'sort by time' when the other user has no achievement, but you do have it unlocked, it duplicates it into unlocked achievements (and then the view transition fails due to duplicate id)

ge-ku commented:
Fixed the bug, now I'm checking if element is where it's needed to be before appending and made those style changes, how does it look now?
Also added avatars for locked achievements with low opacity similar to Steam client.

![image](https://github.com/user-attachments/assets/cae79ac9-bbae-4279-acd9-2f72494f4475)

xPaw commented:
I'll check in a bit. Also need to test if the progress parsing works correctly in other languages.

ge-ku commented:
I did a quick test and it looks like it always returns progress text in the same format for every language.

```
const languages = [
  'arabic', 'bulgarian', 'schinese', 'tchinese', 'czech',
  'danish', 'dutch', 'english', 'finnish', 'french',
  'german', 'greek', 'hungarian', 'indonesian', 'italian',
  'japanese', 'koreana', 'norwegian', 'polish', 'portuguese',
  'brazilian', 'romanian', 'russian', 'spanish', 'latam',
  'swedish', 'thai', 'turkish', 'ukrainian', 'vietnamese',
];

const values = new Set();

for (const lang of languages) {
  const res = await fetch(
    `https://steamcommunity.com/id/ChetFaliszek/stats/477160/compare?l=${lang}`
  );
  const html = await res.text();
  const parser = new DOMParser();
  const dom = parser.parseFromString(html, 'text/html');

  const row = dom
    .querySelector(
      'img[src="https://cdn.akamai.steamstatic.com/steamcommunity/public/images/apps/477160/3dd44d541eaa4d24127e92d0c42f2bbac1a58f64.jpg"]'
    )
    .closest('.achieveRow');
  const progressText = row.querySelector('.progressText').textContent.trim();
  console.log(lang, progressText);
  values.add(progressText);
}

console.log(values);
```

xPaw commented:
Other users' avatar is not muted for their locked achievements.

xPaw commented:
Thanks!

xPaw commented:
Mobile view needs a bit of fixing.
